--- linux-live-5.5.0/config	2006-10-02 20:19:58.000000000 +0300
+++ linux-live-5.5.0/config	2006-09-29 15:30:28.000000000 +0300
@@ -24,12 +24,21 @@
 if [ ! -d $initrddir ]; then
 	echo >&2 "Creating $initrd dir"
 	mkdir -p $initrddir
-	cp -af $ROOT/lib/modules/$KERNEL/kernel/fs/squashfs/squashfs.ko* $initrddir
-	if [ -f $ROOT/lib/modules/$KERNEL/kernel/fs/unionfs.ko* ]; then
-		cp -af $ROOT/lib/modules/$KERNEL/kernel/fs/unionfs.ko* $initrddir
-	else
-		cp -af $ROOT/lib/modules/$KERNEL/kernel/fs/unionfs/unionfs.ko* $initrddir
-	fi
+	for mod in \
+		drivers/cdrom/cdrom \
+		drivers/ide/ide-cd \
+		drivers/ide/ide-core \
+		fs/isofs/isofs \
+		fs/squashfs/squashfs \
+		fs/unionfs \
+		fs/unionfs/unionfs \
+		fs/vfat/vfat \
+		lib/zlib_inflate/zlib_inflate \
+		; do
+		if [ -f $ROOT/lib/modules/$KERNEL/kernel/$mod.ko* ]; then
+			cp -af $ROOT/lib/modules/$KERNEL/kernel/$mod.ko* $initrddir
+		fi
+	done
 fi
 
 # list of directories which will be modularized
--- linux-live-5.5.0/initrd/linuxrc	2006-10-02 20:19:58.000000000 +0300
+++ linux-live-5.5.0/initrd/linuxrc	2006-09-28 01:19:28.000000000 +0300
@@ -185,6 +185,9 @@
 fi
 
 header "linux live end, starting the Linux distribution"
+
+$DEBUGCMD
+
 pivot_root . $INITRAMDISK
 exec $CHROOT . sbin/init <dev/console >dev/console 2>&1
 
--- linux-live-5.5.0/tools/liblinuxlive	2006-10-02 20:19:58.000000000 +0300
+++ linux-live-5.5.0/tools/liblinuxlive	2006-09-29 18:04:58.000000000 +0300
@@ -441,10 +441,13 @@
 #
 modprobe_essential_modules()
 {
+   echolog "starting IDE device support"
+   modprobe_module piix
    echolog "starting loop device support"
    modprobe_module loop max_loop=255
    echolog "starting cdrom filesystem support"
    modprobe_module isofs
+   modprobe_module ide-cd
    echolog "starting squashfs support"
    modprobe_module squashfs
    echolog "starting unionfs support"
@@ -456,6 +459,12 @@
    modprobe_module vfat
    echolog "starting ntfs support"
    modprobe_module ntfs
+   echolog "starting xfs support"
+   modprobe_module xfs
+   echolog "starting reiserfs support"
+   modprobe_module reiserfs
+   echolog "starting ext3 support"
+   modprobe_module ext3
    create_block_devices
 }
 
--- linux-live-5.5.0/cd-root/isolinux.cfg	2006-05-05 23:25:22.000000000 +0300
+++ linux-live-5.5.0/cd-root/isolinux.cfg	2006-10-03 14:03:53.000000000 +0300
@@ -7,7 +7,7 @@
 
 label linux
 kernel boot/vmlinuz
-append vga=769 max_loop=255 initrd=boot/initrd.gz init=linuxrc load_ramdisk=1 prompt_ramdisk=0 ramdisk_size=4444 root=/dev/ram0 rw
+append vga=769 max_loop=255 initrd=boot/initrd.gz init=linuxrc load_ramdisk=1 prompt_ramdisk=0 ramdisk_size=6444 root=/dev/ram0 rw
 
 label memtest
 kernel boot/memtest
