--- linux-live/initrd/initrd_create	2006-10-13 14:38:45.000000000 +0300
+++ linux-live-5.5.0/initrd/initrd_create	2006-10-13 16:36:35.000000000 +0300
@@ -3,11 +3,12 @@
 #
 # Author:	  Tomas M. <http://www.linux-live.org>
 
-. ../config || exit 1
+. /etc/linux-live/config || exit 1
 
 # rcopy is a recursive cp, which copies also symlink's real source
 # $1 = source (may be a regular file or symlink)
 # $2 = target PARENT
+# $3 = optional PREFIX to strip from $1
 #
 rcopy()
 {
@@ -16,7 +17,11 @@
       cp --parent -R "$REALPATH" "$2"
       ln -sf "$REALPATH" "$2/$1"
    else
-      cp --parent -R "$1" "$2"
+	  if [ "$3" ]; then
+		  tar ${3:+-C "$3"} -cf - "${1#$3/}" | tar -C "$2" -xf -
+	  else
+		  cp --parent -R "$1" "$2"
+	  fi
    fi
    if [ "$?" -ne 0 ]; then
       echo "---------------------------"
@@ -31,11 +36,12 @@
 # copy file/dir only if it exists, else skip with no error
 # $1 = source (may not exist)
 # $2 = target PARENT
+# $3 = optional PREFIX to strip from $1
 #
 rcopy_ex()
 {
    if [ -a "$1" ]; then
-      rcopy "$1" "$2"
+      rcopy "$1" "$2" "$3"
    fi
 }
 
@@ -109,40 +115,88 @@
 ln -s busybox $INITRD_TREE/bin/cat
 ln -s busybox $INITRD_TREE/bin/grep
 ln -s busybox $INITRD_TREE/bin/sleep
+cp -a /bin/mawk $INITRD_TREE/bin/awk
 ln -s bin $INITRD_TREE/sbin
+mkdir -p $INITRD_TREE/usr/share
+cp -a /usr/bin/pcidev $INITRD_TREE/bin
+cp -a /usr/share/pci-database $INITRD_TREE/usr/share
+rm -f $INITRD_TREE/lib/*
+if [ -d /lib64 ]; then
+	lib=lib64
+	ld_linux=ld-linux-x86-64.so.?
+else
+	lib=lib
+	ld_linux=ld-linux.so.?
+fi
+rcopy /$lib/$ld_linux $INITRD_TREE
+rcopy /$lib/libc.so.? $INITRD_TREE
+rcopy /$lib/libblkid.so.? $INITRD_TREE
+rcopy /$lib/libuuid.so.? $INITRD_TREE
+rcopy /$lib/libz.so.? $INITRD_TREE
+rcopy /$lib/libm.so.* $INITRD_TREE
+rcopy /$lib/libdevmapper.so.* $INITRD_TREE
+rcopy /$lib/libselinux.so.* $INITRD_TREE
+rcopy /$lib/libsepol.so.* $INITRD_TREE
+rcopy /$lib/libdl.so.* $INITRD_TREE
 
 LMK="lib/modules/$KERNEL"
 
 #necessary modules and dependency files
 mkdir -p $INITRD_TREE/$LMK/kernel/fs
-cp kernel-modules/$KERNEL/unionfs.ko* $INITRD_TREE/$LMK/kernel/fs
-cp kernel-modules/$KERNEL/squashfs.ko* $INITRD_TREE/$LMK/kernel/fs
+#cp kernel-modules/$KERNEL/*.ko* $INITRD_TREE/$LMK/kernel/fs
+#cp kernel-modules/$KERNEL/squashfs.ko* $INITRD_TREE/$LMK/kernel/fs
+#cp kernel-modules/$KERNEL/zlib_inflate.ko* $INITRD_TREE/$LMK/kernel/fs
 
+(
 #copy filesystem modules, if not directly copied into kernel
-rcopy_ex /$LMK/kernel/lib/zlib_inflate $INITRD_TREE 2>>$DEBUG
-rcopy_ex /$LMK/kernel/lib/zlib_deflate $INITRD_TREE 2>>$DEBUG
-rcopy_ex /$LMK/kernel/drivers/block/loop* $INITRD_TREE 2>>$DEBUG
-
-rcopy_ex /$LMK/kernel/fs/isofs $INITRD_TREE 2>>$DEBUG
-rcopy_ex /$LMK/kernel/fs/fat $INITRD_TREE 2>>$DEBUG
-rcopy_ex /$LMK/kernel/fs/vfat $INITRD_TREE 2>>$DEBUG
-rcopy_ex /$LMK/kernel/fs/ntfs $INITRD_TREE 2>>$DEBUG
-rcopy_ex /$LMK/kernel/fs/ext3 $INITRD_TREE 2>>$DEBUG
-rcopy_ex /$LMK/kernel/fs/reiserfs $INITRD_TREE 2>>$DEBUG
+rcopy $ROOT/$LMK/kernel/lib/zlib_inflate $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/lib/zlib_deflate $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/block/loop.ko* $INITRD_TREE $ROOT
+
+rcopy_ex $ROOT/$LMK/kernel/fs/isofs $INITRD_TREE $ROOT
+rcopy_ex $ROOT/$LMK/kernel/fs/isofs.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/fs/fat $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/fs/vfat $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/fs/ntfs $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/fs/ext3 $INITRD_TREE $ROOT
+#rcopy $ROOT/$LMK/kernel/fs/xfs $INITRD_TREE $ROOT # causes weird issues
+rcopy $ROOT/$LMK/kernel/fs/exportfs $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/fs/reiserfs $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/fs/jbd/jbd.ko* $INITRD_TREE $ROOT
 
 # add language support for filesystems
-rcopy_ex /$LMK/kernel/fs/nls/ $INITRD_TREE 2>>$DEBUG
+rcopy $ROOT/$LMK/kernel/fs/nls/nls_cp437.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/fs/nls/nls_iso8859-1.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/fs/nls/nls_iso8859-2.ko* $INITRD_TREE $ROOT
 
 #usb modules
-rcopy_ex /$LMK/kernel/drivers/usb/storage $INITRD_TREE 2>>$DEBUG
-rcopy_ex /$LMK/kernel/drivers/usb/host/ehci-hcd* $INITRD_TREE 2>>$DEBUG
-rcopy_ex /$LMK/kernel/drivers/usb/host/ohci-hcd* $INITRD_TREE 2>>$DEBUG
-rcopy_ex /$LMK/kernel/drivers/usb/host/uhci-hcd* $INITRD_TREE 2>>$DEBUG
+rcopy $ROOT/$LMK/kernel/drivers/usb/core/usbcore.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/usb/input/usbhid.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/usb/storage $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/usb/host/ehci-hcd.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/usb/host/ohci-hcd.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/usb/host/uhci-hcd.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/usb/storage/libusual.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/scsi/scsi_mod.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/scsi/sr_mod.ko* $INITRD_TREE $ROOT
+
+rcopy $ROOT/$LMK/kernel/drivers/cdrom/cdrom.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/ide/ide-cd.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/ide/ide-core.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/ide/pci/piix.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/scsi/ata_piix.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/drivers/scsi/libata.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/fs/isofs/isofs $INITRD_TREE $ROOT
+rcopy_ex $ROOT/$LMK/kernel/fs/squashfs.ko* $INITRD_TREE $ROOT
+rcopy_ex $ROOT/$LMK/kernel/fs/squashfs $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/fs/unionfs.ko* $INITRD_TREE $ROOT
+rcopy $ROOT/$LMK/kernel/fs/unionfs/unionfs $INITRD_TREE $ROOT
 
 #disk (scsi, ide, raid, pcmcia) modules
 #rcopy_ex /$LMK/kernel/drivers/scsi $INITRD_TREE
 #rcopy_ex /$LMK/kernel/drivers/ide $INITRD_TREE
 #rcopy_ex /$LMK/kernel/drivers/pcmcia $INITRD_TREE
+) 2>>$DEBUG
 
 if [ "$COMPRESS" = 1 ]; then
    debug "gzipping kernel modules"
--- _broken-linux-live-5.5.0/initrd/linuxrc	2006-09-26 18:37:04.000000000 +0300
+++ linux-live-5.5.0/initrd/linuxrc	2006-09-28 01:19:28.000000000 +0300
@@ -157,5 +157,8 @@
 fstab_update $UNION
 
+echolog "creating /etc/modprobe.conf"
+modprobe_update $UNION
+
 # More likely these directories aren't there.
 # Even if they are, this won't hurt.
 # Even if they are, this won't hurt.
--- _broken-linux-live-5.5.0/tools/liblinuxlive	2006-09-26 17:18:28.000000000 +0300
+++ linux-live-5.5.0/tools/liblinuxlive	2006-09-28 03:14:39.000000000 +0300
@@ -367,6 +367,13 @@
    list_partition_devices
 }
 
+# List all network drivers
+#
+list_network_drivers()
+{
+	pcidev /m net
+}
+
 # Find file-path on given device
 # Mounts the device in $1 and returns path if found,
 # else unmounts and exits
@@ -561,0 +569,13 @@
+
+# create modprobe.conf file $1/etc/modprobe.conf with appropriate ethX module aliases
+# $1 = root directory (union)
+#
+modprobe_update()
+{
+	i=0
+	> $1/etc/modprobe.conf
+	for drv in $(list_network_drivers); do
+		echo "alias eth$i $drv" >> $1/etc/modprobe.conf
+		i=$((i+1))
+	done
+}
